### PR TITLE
feat: wallet create functionality

### PR DESCRIPTION
**Pull request by alexbarnsley**
## Summary

Note: I've omitted download because that requires electron which currently has issues with cypress (#2485)

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: src/app/components/SelectNetwork/SelectNetwork.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ type SelectNetworkProps = {
 	networks: Network[];
 	placeholder?: string;
 	name?: string;
-	value?: string;
+	value?: any;
```
Comment (faustbrian) [inline src/app/components/SelectNetwork/SelectNetwork.tsx:18]: Why was this changed to `any`?

File: src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx [hunk 2]
```diff
@@ -1,27 +1,57 @@
...
 
 describe("CreateWallet", () => {
+	beforeEach(async () => {
+		env = new Environment({
+			coins: { ARK },
+			httpClient,
+			storage: new StubStorage(),
+		});
+
+		await env.boot();
```
Comment (faustbrian) [inline src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx:43]: This could be replaced with https://github.com/ArkEcosystem/platform-sdk/commit/031c6008a3f3de3f85349550e6300873a589fbc0 but also works for now as is.

File: src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx [hunk 3]
```diff
@@ -32,36 +62,53 @@ describe("CreateWallet", () => {
...
-			fireEvent.change(selectAssetsInput, { target: { value: "Bitco" } });
+			fireEvent.change(selectAssetsInput, { target: { value: "ARK" } });
 		});
 
 		await act(async () => {
 			fireEvent.keyDown(selectAssetsInput, { key: "Enter", code: 13 });
 		});
 
-		expect(getByTestId("select-asset__selected-Bitcoin")).toBeTruthy();
+		expect(getByTestId("select-asset__selected-ARK - Mainnet")).toBeTruthy();
```
Comment (faustbrian) [inline src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx:72]: IDs and Classes shouldn't contain whitespace characters.

Comment (brenopolanski) [reply src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx:72]: @alexbarnsley I will create a separate PR to fix this in `SelectNetwork`, so I can update my PR as well.

File: src/app/components/SelectNetwork/SelectNetwork.tsx (hunk 1 above)
Comment (alexbarnsley) [reply src/app/components/SelectNetwork/SelectNetwork.tsx:18]: I pass in the Network object itself, not a string

File: src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx (hunk 2 above)
Comment (alexbarnsley) [reply src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx:43]: Will update, thanks 👌

File: src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx (hunk 3 above)
Comment (alexbarnsley) [reply src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx:72]: thanks @brenopolanski 👌 I agree it shouldn't contain whitespace

File: src/app/components/SelectNetwork/SelectNetwork.tsx (hunk 1 above)
Comment (faustbrian) [reply src/app/components/SelectNetwork/SelectNetwork.tsx:18]: Shouldn't it then be changed to `object`, `Record<string, unknown>` or some specific object?

Comment (alexbarnsley) [reply src/app/components/SelectNetwork/SelectNetwork.tsx:18]: 🤔 I'll see how it's used elsewhere

File: src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx [hunk 4]
```diff
@@ -1,27 +1,65 @@
...
 
 describe("CreateWallet", () => {
+	beforeEach(async () => {
+		env = new Environment({
+			coins: { ARK },
+			httpClient,
+			storage: new StubStorage(),
+		});
+
+		await env.bootFromObject({ profiles });
```
Comment (brenopolanski) [inline src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx:47]: There is a warning here, change to `await env.bootFromObject({ data: {}, profiles });`, because the data is required in type `StorageData`.

File: src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx [hunk 5]
```diff
@@ -141,38 +184,143 @@ describe("CreateWallet", () => {
...
+				{
+					routes: [createURL, "/"],
+					history,
+				},
+			);
+
+			await waitFor(() => expect(rendered.getByTestId(`CreateWallet__first-step`)).toBeTruthy());
+		});
+
+		const { getByTestId, getByText, asFragment } = rendered!;
```
Comment (brenopolanski) [inline src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx:298]: `getByText` is declared but its value is never read.

File: src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx (hunk 4 above)
Comment (alexbarnsley) [reply src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx]: Updated, thanks!

File: src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx (hunk 5 above)
Comment (alexbarnsley) [reply src/domains/wallet/pages/CreateWallet/CreateWallet.test.tsx]: Updated, thanks!

File: src/domains/wallet/pages/CreateWallet/CreateWallet.tsx [hunk 6]
```diff
@@ -214,29 +259,22 @@ export const CreateWallet = ({ networks, mnemonic, onSubmit, onCopy, onDownload,
...
-			route: `/profiles/${activeProfile?.id()}/dashboard`,
-			label: "Go back to Portfolio",
-		},
-	];
-
 	return (
 		<Page crumbs={crumbs}>
 			<Section className="flex-1">
-				<Form className="max-w-xl mx-auto" context={form} onSubmit={(data: any) => onSubmit(data)}>
+				<Form className="max-w-xl mx-auto" context={form} onSubmit={(data: any) => submitForm(data)}>
```
Comment (brenopolanski) [inline src/domains/wallet/pages/CreateWallet/CreateWallet.tsx:265]: Change the `onSubmit` to `onSubmit={submitForm}>`

Comment (alexbarnsley) [reply src/domains/wallet/pages/CreateWallet/CreateWallet.tsx]: done :bleh:
